### ISSUE TITLE
Adding nested keys in JSON format does not work

### ISSUE THREAD

**Issue by tobiasgiese** [getsops/sops#801]
# Bug Report 

Adding nested keys in JSON format is unfortunately not possible.
I am using the latest version v3.6.1.

What I am trying:

```
$ sops --version
sops 3.6.1 (latest)
$ echo '{"foo":{"bar":{"key":"foobar"}},"unrelated":"value"}' > foobar.json
$ sops -e -i --input-type json ./foobar.json
$ sops -d foobar.json
{
	"foo": {
		"bar": {
			"key": "foobar"
		}
	},
	"unrelated": "value"
}
$ sops --set '["foo"]["baz"]["new"] "value"' ./foobar.json
$ sops -d foobar.json
{
	"foo": {
		"baz": {
			"new": "value"
		}
	},
	"unrelated": "value"
}
```

What I expected to happen:

```json
{
	"foo": {
		"bar": {
			"key": "foobar"
		},
		"baz": {
			"new": "value"
		}
	},
	"unrelated": "value"
}
```

<sup>
Tobias Giese <tobias.giese@daimler.com>, Daimler TSS GmbH, 
<a href="https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md">Imprint</a>
</sup>

**Comment by samrees** [comment]
it deletes everything under the existing key. still exists in 3.7.3

**Comment by fangpenlin** [comment]
I ran into the same problem, but it appears that the bug has been addressed by #899 and released as 3.8.0

**Comment by felixfontein** [comment]
Thanks!

Closing since fixed in #899.
